### ISSUE TITLE
uninstall command for vsix (visual studio extension)

### ISSUE THREAD

**Issue by timabell** [chocolatey/choco#389]
Hi, hope you don't mind me asking here. I'm trying to create a package for a visual studio extension (vsix). I've got the install working; now I am wondering about allowing uninstall, it would be nice if I could provide that but there doesn't seem to provision for it.

I was following the instructions at https://github.com/chocolatey/choco/wiki/HelpersInstallChocolateyVsixPackage but can't find an equivalent uninstall command.

`C:\Program Files (x86)\Microsoft Visual Studio 14.0>vsixinstaller /?` does show an uninstall switch.

I assume it's just not implemented? Cheers

Thanks for a great tool.

**Comment by ferventcoder** [comment]
@timabell You are correct, an uninstall helper just has not been added. If you want to add one, we accept pull requests. Just make sure you read over our Contributing doc prior to ensure you set up commits correctly and can sign the CLA. :)

**Comment by timabell** [comment]
Okay thanks for confirming. I'll pass on the PR then as I'm not keen on CLAs.

**Comment by ferventcoder** [comment]
No worries.

**Comment by kevindaub** [comment]
I was going to look into this one.  At first glance, it looks like the uninstall and install will have very similar code (i.e. like locating VS Install directory).

* Would creating another function like Get-VisualStudioInstallDir or Get-VisualStudioVersion be okay to add as a helper function as well?
* Also, are there any powershell unit tests or integration tests that should be included (I see C#, but no powershell)?

**Comment by ferventcoder** [comment]
C# integration tests are in for testing different behaviors of packages, installing and uninstalling, but don't actually test making changes to the system itself. I've been thinking about how to introduce those kinds of tests without having to change system state. You are welcome to make suggestions. 

All of the PowerShell functions will be reimplemented as C# PowerShell cmdlets at some point, so something else to consider.

**Comment by kevindaub** [comment]
I'd prefer to keep in C#, so maybe that's the path I'll go down.  It also allows easier testing.

**Comment by pauby** [comment]
Thanks for raising this issue. After reviewing it, it's not something we expect to prioritise in the foreseeable future. We’re closing it to keep our backlog focused, but we’re open to revisiting it in the future if priorities shift or new interest arises. You are welcome to comment here or open a new issue if you'd like to discuss it further down the line.
